src/lib.rs

```diff
@@ -220,10 +223,35 @@ cfg_if! {
     if #[cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))] {
         mod util_libc;
         #[path = "use_file.rs"] mod imp;
-    } else if #[cfg(any(target_os = "android", target_os = "linux"))] {
+    } else if #[cfg(all(
+        not(feature = "disable_urandom_fallback"),
+        any(
+            // Rust supports Android API level 19 (KitKat) [0], while `getrandom(2)`
+            // was added only in level 23 (Marshmallow).
+            // [0]: https://blog.rust-lang.org/2023/01/09/android-ndk-update-r25.html
+            target_os = "android",
```

Comment:
I think this should also be conditional on target_arch in the same way that it is for "linux", except it should also include "aarch64" for Android only.

For example, RISC-V Android would never (I hope) ship with a kernel so old that it would need this fallback.

Reply:
To be safe I've also added `x86` and `x86_64` to this list.